crates/nu-plugin-protocol/src/lib.rs

```diff
@@ -57,6 +59,32 @@ pub struct CallInfo<D> {
     pub input: D,
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum GetCompletionArgType {
+    Flag(String),
+    Positional(usize),
+}
+
+impl<'a> From<GetCompletionArgType> for ArgType<'a> {
+    fn from(value: GetCompletionArgType) -> Self {
+        match value {
+            GetCompletionArgType::Flag(flag_name) => {
+                ArgType::Flag(std::borrow::Cow::from(flag_name))
+            }
+            GetCompletionArgType::Positional(idx) => ArgType::Positional(idx),
+        }
+    }
+}
+
+/// Information about `get_dynamic_completion` of a plugin call invocation.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct GetCompletionInfo {
+    /// The name of the command to be run.
+    pub name: String,
+    /// The flag name to get completion items.
+    pub arg_type: GetCompletionArgType,
```

Comment:
Why do we need a new `GetCompletionArgType` struct? Why not reuse `ArgType`?

Reply:
It's because `GetCompletionArgType` needs to derive `Serialize, Deserialize` in order to communicate with plugin.
But I want `ArgType` to be defined with `Cow<str>`, so nushell can get rid of unnecessary allocation for string, when it want to do flag completion.  If I really want to make the `ArgType` derive `Serialize, Deserialize`, I need to use [BorrowCow in serde_with](https://docs.rs/serde_with/latest/serde_with/struct.BorrowCow.html) crate, which is overkill.